target/linux/generic/files/drivers/net/phy/rtl8366_smi.c

```diff
@@ -919,6 +969,21 @@ static int rtl8366_smi_mii_init(struct rtl8366_smi *smi)
 {
 	int ret;
 	int i;
+	struct mii_bus *bus;
+	struct device_node *np;
+
+	if (smi->mdio_enabled) {
+		np = of_parse_phandle(smi->parent->of_node, "mdio", 0);
+
+		if (np) {
+			bus = of_mdio_find_bus(np);
+
+			if (bus) {
+				smi->mii_bus = bus;
+				return 0;
+			} else return -ENODEV;
```

Comment:
I tested the patchset on a lantiq board, which is not yet supported because of the missing rtl8367b mdio support.

This return has to be -EPROBE_DEFER. In my case the mdio driver is loaded after the switch driver. Means, at the time the switch driver is loaded, there is no mdio bus yet. If you use -EPROBE_DEFER the switch driver is probed again at a later time.

The same used in the kernel in conjunction with of_mdio_find_bus as well.